### PR TITLE
gh-126980: Fix `bytearray.__buffer__` crash on `PyBUF_{READ,WRITE}`

### PR DESCRIPTION
**Pull request by sobolevn**
Related https://github.com/python/cpython/pull/114802

@JelleZijlstra, since you are `__buffer__` PEP author, could you please take a look? :)

* Issue: gh-126980

### CODE REVIEW

File: Lib/test/test_buffer.py [hunk 1]
```diff
@@ -4439,6 +4439,13 @@ def test_issue_7385(self):
         x = ndarray([1,2,3], shape=[3], flags=ND_GETBUF_FAIL)
         self.assertRaises(BufferError, memoryview, x)
 
+    def test_bytearray_release_buffer_read_flag(self):
+        # See https://github.com/python/cpython/issues/126980
+        with self.assertRaises(SystemError):
+            bytearray().__buffer__(inspect.BufferFlags.READ)
+        with self.assertRaises(SystemError):
+            bytearray().__buffer__(inspect.BufferFlags.WRITE)
```
Comment (vstinner) [inline Lib/test/test_buffer.py]: Maybe use an non-empty bytearray:

```suggestion
        # See https://github.com/python/cpython/issues/126980
        obj = bytearray(b'abc')
        with self.assertRaises(SystemError):
            obj.__buffer__(inspect.BufferFlags.READ)
        with self.assertRaises(SystemError):
            obj.__buffer__(inspect.BufferFlags.WRITE)
```

**Comment by JelleZijlstra** [comment]
> @JelleZijlstra, since you are `__buffer__` PEP author, could you please take a look? :)

I don't think I wrote `bytearray_getbuffer`; the buffer protocol predates my PEP. This change looks good to me though, thanks!

File: Lib/test/test_buffer.py (hunk 1 above)
Comment (sobolevn) [reply Lib/test/test_buffer.py]: Ok!

**Review by skirpichev — APPROVED** [review]

**Review by vstinner — APPROVED** [review]

**Comment by sobolevn** [comment]
This needs backport to 3.13: https://github.com/python/cpython/blob/3.13/Objects/abstract.c#L770-L774
But not to 3.12: https://github.com/python/cpython/blob/3.12/Objects/abstract.c#L718

**Comment by sobolevn** [comment]
Will do a manual one.

**Pull request merged**